Use env var for auth frontend

tarzzz opened the pull request:
- [x] Test on desktop app
  - [x] Test on on-prem

tarzzz commented:
@n-riesco Please review..

n-riesco commented on app/components/Configuration.react.js:
```diff
@@ -14,7 +14,8 @@ class Configuration extends Component {
             isMenuOpen: false,
             username: cookie.load('db-connector-user'),
-            authDisabled: cookie.load('db-connector-auth-disabled')
+            // eslint-disable-next-line no-undef
```
use `/*global PLOTLY_ENV*/` instead

n-riesco commented on webpack.config.web.js:
```diff
@@ -23,6 +23,11 @@ const config = {
             __DEV__: false,
             'process.env.NODE_ENV': JSON.stringify('production')
+        }),
+        new webpack.DefinePlugin({
```
Please, add a comment explaining why this is needed.

n-riesco commented:
@tarzzz This PR does what you explained in [solution 1](https://github.com/plotly/streambed/issues/10436#issuecomment-351064663). How about @scjody 's [solution](https://github.com/plotly/streambed/issues/10436#issuecomment-351122646)? Wouldn't it be easier and avoid the build step at the ENTRYPOINT?

n-riesco commented on webpack.config.web.js:
```diff
@@ -24,6 +24,10 @@ const config = {
+        /*
+         * This is used to pass environment variables to frontend when docker
+         * container initializes
+         */
```
Sorry for being a pain, but this is not enough. We'll forget very soon. Please, could you at least add a link to the streambed issue?

n-riesco commented on app/components/Configuration.react.js:
```diff
@@ -14,7 +14,7 @@ class Configuration extends Component {
             isMenuOpen: false,
             username: cookie.load('db-connector-user'),
-            // eslint-disable-next-line no-undef
+            /*global PLOTLY_ENV*/
```
linter fails here, spacing needed: `/* global PLOTLY_ENV */`

n-riesco commented on webpack.config.web.js:
```diff
@@ -2,6 +2,9 @@ import webpack from 'webpack';
 import baseConfig from './webpack.config.base';
 import path from 'path';
 
+const AUTH_ENABLED = (process.env.PLOTLY_CONNECTOR_AUTH_ENABLED !== 'undefined' ?
```
This doesn't work. How about:
```
const AUTH_ENABLED = process.env.PLOTLY_CONNECTOR_AUTH_ENABLED ?
    JSON.parse(process.env.PLOTLY_CONNECTOR_AUTH_ENABLED) :
    true;
```
?

tarzzz commented:
@n-riesco Addressed the comments.. Please let me know if it looks good.

I have also added a `OAUTH2_CLIENT_ID` variable, which is to replace the hardcoded `clientId` in `Login.react.js`. Apart from that, this is also needed for another on-premise issue.

n-riesco commented on webpack.config.web.js:
```diff
@@ -2,8 +2,12 @@ import webpack from 'webpack';
-const AUTH_ENABLED = (process.env.PLOTLY_CONNECTOR_AUTH_ENABLED !== 'undefined' ?
-                      JSON.parse(process.env.PLOTLY_CONNECTOR_AUTH_ENABLED): true);
+const AUTH_ENABLED = process.env.PLOTLY_CONNECTOR_AUTH_ENABLED ?
+                     JSON.parse(process.env.PLOTLY_CONNECTOR_AUTH_ENABLED) : true;
```
:cow: the rest of the code uses 4-space indent

n-riesco commented on webpack.config.web.js:
```diff
@@ -2,8 +2,12 @@ import webpack from 'webpack';
+
+const OAUTH2_CLIENT_ID = process.env.PLOTLY_CONNECTOR_OAUTH2_CLIENT_ID ?
+                         JSON.stringify(process.env.PLOTLY_CONNECTOR_OAUTH2_CLIENT_ID) :
+                         JSON.stringify('isFcew9naom2f1khSiMeAtzuOvHXHuLwhPsM7oPt');
```
:cow: the rest of the code uses 4-space indent

n-riesco commented:
This PR is OK as a quick fix for the onprem release, but we need to find an alternative solution straight after release.

@scjody , are you OK with having the cliend ID stored as a global for this quick fix?

tarzzz commented:
> are you OK with having the cliend ID stored as a global for this quick fix?

More info, this is related to consent screen bug (https://github.com/plotly/streambed/issues/10444), where we can provide custom client ID as an On Prem Environment Variable. For non-prem, the client ID will still point to https://plot.ly/

scjody commented:
> are you OK with having the cliend ID stored as a global for this quick fix?

@n-riesco It's stored as a global in the current version. What's your new concern here?

n-riesco commented:
@scjody it's stored in local variable [here](https://github.com/plotly/falcon-sql-client/blob/0568b760aac7f8cfb372d20bb70729c63894ada1/app/components/Login.react.js#L129). I may be wrong, but I don't think Falcon uses global variables (by global variable I mean a variable defined in the global context: `window` in the browser and `global` in Node).

scjody commented:
Seems reasonable to me. It's not a confidential value; anyone who looks at this repo can see it. (OAuth security in our case depends on the `redirect_uri` being something that an attacker can't serve data from, not from keeping client parameters secret.)

tarzzz commented:
Addressed the 🐮 ..

@n-riesco 💃 ??

n-riesco commented:
:dancer:
